x-pack/plugin/core/build.gradle

```diff
@@ -39,6 +39,16 @@ dependencies {
     // security deps
     shadow 'com.unboundid:unboundid-ldapsdk:3.2.0'
     shadow project(path: ':modules:transport-netty4', configuration: 'runtime')
+    shadow(project(path: ':plugins:transport-nio', configuration: 'runtime')) {
+        exclude group: "org.elasticsearch", module: "elasticsearch-core"
```

Comment:
This looks like a bug in esplugin, that core is not being marked as compileOnly in this case (yet it is for depending on the transport-netty4 module). @atorok can you investigate this and open an issue? @tbrooks8 can you add a TODO/comment here that this should not be necessary